test/python/quantum_info/operators/test_measures.py

```diff
@@ -178,11 +180,33 @@ def test_diamond_norm(self, num_qubits):
             op = op + coeff * Choi(Operator.from_label(label))
         target = np.sum(np.abs(coeffs))
 
-        try:
-            value = diamond_norm(op)
-            self.assertAlmostEqual(value, target, places=4)
-        except cvxpy.SolverError:
-            self.skipTest("CVXPY solver failed.")
+        value = diamond_norm(op)
+        self.assertAlmostEqual(value, target, places=4)
+
+    def test_unitary_diamond_distance(self):
+        """Test the unitary_diamond_distance function for RZGates
+        with a specific set of angles."""
+        angles = np.linspace(0, 2 * np.pi, 10, endpoint=False)
+        for angle in angles:
+            op1 = Operator(RZGate(angle))
+            op2 = Operator.from_label("I")
+            d2 = np.cos(angle / 2) ** 2  # analytical formula for hull distance
+            target = np.sqrt(1 - d2) * 2
+            self.assertAlmostEqual(unitary_diamond_distance(op1, op2), target, places=7)
+
+    @combine(num_qubits=[1, 2, 3])
+    def test_unitary_diamond_distance_random(self, num_qubits):
+        """Tests the unitary_diamond_distance for random unitaries.
+        Compares results with semi-definite program."""
+        if importlib.util.find_spec("cvxpy") is None:
+            # Skip test if CVXPY not installed
+            self.skipTest("CVXPY not installed.")
+
+        rng = np.random.default_rng(1234)
+        op1 = random_unitary(2**num_qubits, seed=rng)
```

Comment:
On my machine it takes around 5.22s to evaluate the diamond norm for 3 qubit channels. Maybe we can run over a bunch of unitaries when num_qubits < 3 and only run one test otherwise?

Reply:
Running the random tests for 1 and 2 qubit only sounds fine to me